beets/ui/commands.py

```diff
@@ -960,7 +974,25 @@ def import_func(lib, opts, args):
     else:
         query = None
         paths = args
-        if not paths:
+
+        # The paths from the logfiles go into a separate list to allow handling
+        # errors differently from user-specified paths.
+        paths_from_logfiles = []
+        for logfile in opts.from_logfiles or []:
+            try:
+                paths_from_logfiles.extend(
+                    list(_paths_from_logfile(syspath(normpath(logfile))))
+                )
+            except ValueError as err:
+                raise ui.UserError('malformed logfile {}'.format(
+                    util.displayable_path(logfile),
+                )) from err
```

Comment:
For this one, at least, perhaps it would be nice to include some evidence about why the log file is malformed? Because `UserError` just prints the associated string by default (and doesn't display the "chained" exception), just including `str(err)` in the message could make a difference in usability.

Reply:
https://github.com/beetbox/beets/pull/4387/commits/cfc34826a20203240e1bc8f2ba0cee328dfe18df